### PR TITLE
Make the Binary Cross Entropy Loss able to handle probabilities as input.

### PR DESCRIPTION
**Pull request by AtomicVar**
## Proposed changes

Currently, the `binary_cross_entropy()` in MLX can only handle logits as input. In PyTorch, there are two seperate binary cross entropy loss:

1. [torch.nn.functional.binary_cross_entropy — PyTorch 2.1 documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy.html)
2. [torch.nn.functional.binary_cross_entropy_with_logits — PyTorch 2.1 documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy_with_logits.html)

The `binary_cross_entropy()` in MLX is actually the `binary_cross_entropy_with_logits()` in PyTorch. To enable the non-logit version, this PR adds an extra (optional) parameter `with_logits` to `binary_cross_entropy()`. If this parameter is given `False`, the input is regarded as probabilities.

To make this change backward-compatible, the `with_logits` is given a default value `True`, thus existing user code don't need to change.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/mlx/nn/losses.py [hunk 1]
```diff
@@ -74,29 +74,50 @@ def cross_entropy(
...
 
     Args:
-        logits (array): The unnormalized (pre-sigmoid) predicted logits.
+        inputs (array): The predicted values. If ``with_logits`` is ``True``, then
+            ``inputs`` are unnormalized logits. Otherwise, ``inputs`` are probabilities.
         targets (array): The binary target values in {0, 1}.
+        with_logits (bool, optional): Whether ``inputs`` are logits. Default: ``True``.
         reduction (str, optional): Specifies the reduction to apply to the output:
-          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'none'``.
+          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'mean'``.
```
Comment (AtomicVar) [inline python/mlx/nn/losses.py:91]: I also changed the default reduction to `mean`, which is the default reduction of PyTorch.

**Review by awni — APPROVED** [review]
Looks good, thanks!
